components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissionsFeature.kt

```diff
@@ -116,27 +180,31 @@ class SitePermissionsFeature(
     @Suppress("MaxLineLength")
     override fun onPermissionsResult(permissions: Array<String>, grantResults: IntArray) {
         sessionManager.runWithSessionIdOrSelected(sessionId) { session ->
-            session.appPermissionRequest.consume { permissionsRequest ->
 
+            val appPermissionRequest =
+                store.state.findTabOrCustomTabOrSelectedTab(customTabId)?.content?.appPermissionRequestsList?.find {
```

Comment:
nit: As this `store.state.findTabOrCustomTabOrSelectedTab(customTabId)?.content?` repeats in the code maybe we can have a helper function. What do you think?